src/include/miopen/solver/implicitgemm_ck_util.hpp

```diff
@@ -114,15 +132,33 @@ ConvSolution InitInvokerFactoryNHWC(const ExecutionContext&,
     }
 
     ConvSolution result;
-    result.invoker_factory =
-        [ck_args     = CKArgsType{problem},
-         sh_conv_ptr = std::shared_ptr{std::move(*ptr_iter)}](const std::vector<Kernel>&) mutable {
-            return [ck_args = std::move(ck_args), sh_conv_ptr = std::move(sh_conv_ptr)](
-                       const Handle& handle, const AnyInvokeParams& primitive_parameters) {
-                const auto& data_ctx = primitive_parameters.CastTo<CastType>();
-                auto argument_ptr    = ck_args.MakeArgPtr(sh_conv_ptr, data_ctx.tensors);
-                auto invoker_ptr     = sh_conv_ptr->MakeInvokerPointer();
+    result.invoker_factory = [ck_args     = CKArgsType{problem},
+                              sh_conv_ptr = std::shared_ptr{std::move(*ptr_iter)}](
+                                 const std::vector<Kernel>&) mutable {
+        return [ck_args = std::move(ck_args), sh_conv_ptr = std::move(sh_conv_ptr)](
+                   const Handle& handle, const AnyInvokeParams& primitive_parameters) {
+            const auto& data_ctx = primitive_parameters.CastTo<CastType>();
+            auto argument_ptr    = ck_args.MakeArgPtr(sh_conv_ptr, data_ctx.tensors);
+            auto invoker_ptr     = sh_conv_ptr->MakeInvokerPointer();
+
+            if constexpr(std::is_same<CastType, miopen::conv::WrWInvokeParams>::value)
+            {
+                auto zero           = 0.0f;
+                const auto& tensors = data_ctx.tensors;
+                SetTensor(handle, tensors.dwDesc, tensors.dw, &zero);
```

Comment:
Let's restructure the code to avoid code duplication:
```C++
{
    Profiler t; // see the Recommendation above
    if(WrW)
    {
        ...set output buffer to 0...
    }
    ....run CK kernel....
} // end of the scope - implicitly destructs t
```
So we do not need to create a copy of the code that launches the CK kernel..